src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -403,9 +403,14 @@ def upload_stream(
             else:
                 raise ex
 
-    def notify_query_listeners(self, query_record: QueryRecord) -> None:
+    def notify_query_listeners(
+        self, query_record: QueryRecord, is_error: bool = False
+    ) -> None:
         with self._lock:
             for listener in self._query_listener:
+                # if listener is not set to record error query, skip
+                if is_error and not listener.include_error:
+                    continue
```

Comment:
```suggestion
                if is_error and not getattr(listener, "include_error", False):
                    continue
```
we do this so that customer listeners do not break.